src/time.rs

```diff
@@ -0,0 +1,114 @@
+/// A type that can measure and format the current time.
+///
+/// This trait is used by [HierarchicalLayer] to include a timestamp with each
+/// [Event] when it is logged.
+///
+/// Notable default implementations of this trait are [LocalDateTime] and `()`.
+/// The former prints the current time as reported by [time's OffsetDateTime]
+/// (note that it may panic! make sure to check out the docs for the [LocalDateTime]),
+/// and the latter does not print the current time at all.
+///
+/// Inspired by the [FormatTime] trait from [tracing-subscriber].
+///
+/// [HierarchicalLayer]: crate::HierarchicalLayer
+/// [Event]: tracing_core::Event
+/// [time's OffsetDateTime]: time::OffsetDateTime
+/// [FormatTime]: tracing_subscriber::fmt::time::FormatTime
+/// [tracing-subscriber]: tracing_subscriber
+// NB:
+//   We can't use `tracing_subscriber::fmt::format::Writer`
+//   since it doesn't have a public constructor.
+pub trait FormatTime {
+    fn format_time(&self, w: &mut impl std::fmt::Write) -> std::fmt::Result;
+}
+
+////////////////////////////////////////////////////////////////////////////////////////////////////
+
+/// Default do-nothing time formatter.
+impl FormatTime for () {
+    fn format_time(&self, _w: &mut impl std::fmt::Write) -> std::fmt::Result {
+        Ok(())
+    }
+}
+
+////////////////////////////////////////////////////////////////////////////////////////////////////
+
+/// Retrieve and print the current wall-clock time in UTC timezone.
+#[derive(Debug, Clone, Copy, Eq, PartialEq, Default)]
+pub struct UtcDateTime;
+
+impl FormatTime for UtcDateTime {
+    fn format_time(&self, w: &mut impl std::fmt::Write) -> std::fmt::Result {
+        let time = time::OffsetDateTime::now_utc();
+        write!(w, "{} {}", time.date(), time.time())
+    }
+}
```

Comment:
Since these impls are fairly independent from everything else, the time crate could indeed be made an optional dependency and these impls `cfg` gated. Do you have an opinion on that?

Reply:
Seems reasonable to me, will do.

Reply:
Done.